firefox_ui_tests/update/test_direct_update.py

```diff
@@ -0,0 +1,58 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from firefox_ui_harness import UpdateTestCase
+
+
+class TestDirectUpdate(UpdateTestCase):
+
+    def setUp(self):
+        UpdateTestCase.setUp(self, is_fallback=False)
+
+    def tearDown(self):
+        try:
+            self.windows.close_all([self.browser])
+        finally:
+            UpdateTestCase.tearDown(self)
+
+    def test_update(self):
+        # Check if the user has permissions to run the update
+        self.assertTrue(self.software_update.allowed)
+
+        try:
+            # Open the about window and check for updates
+            about_window = self.browser.open_about_window()
+            update_available = about_window.check_for_updates()
+            self.assertTrue(update_available)
+
+            # Download update and wait until it has been applied
+            about_window.download()
+            about_window.wait_for_update_applied()
+
+        finally:
+            self.updates[self.current_update_index]['patch'] = about_window.patch_info
+
```

Comment:
I can combine that for sure. Just wanted to see what could be necessary for multiple update steps. But once we work on those we can split the shared code where necessary.